src/ContainerManagement.js

```diff
@@ -1,35 +1,60 @@
 const _forEach = require('lodash.foreach');
+const _get = require('lodash.get');
+const _isFunction = require('lodash.isfunction');
+const _isObject = require('lodash.isobject');
 const Dependency = require('./Dependency');
 
 const rall = /.+/;
 
 module.exports = {
 
-  warnIfNeeded(name) {
+  getDependencySourceHint(dependency) {
+    if (_isFunction(dependency)) {
+      return _get(dependency, 'name') || '<anonymous function>';
+    }
+
+    if (_isObject(dependency)) {
+      return _get(dependency, 'constructor.name') || '<object>';
+    }
+
+    return '<primitive>';
```

Comment:
Can we print the type? May help with the debugging.

Reply:
Updated 👍